detox/src/devices/drivers/android/genycloud/services/GenyInstanceLifecycleService.js

```diff
@@ -0,0 +1,25 @@
+const Instance = require('./dto/GenyInstance');
+
+class GenyInstanceLifecycleService {
```

Comment:
This class looks quite thin — I barely see a reason why this class cannot be turned into private methods of `GenyAllocationHelper`. Too much s|p|l|i|t|t|i|n|g, to my taste. 😒  Every entity in the chain GenyCloudDriver ← GenyAllocationHelper ← (GenyInstanceLifecycleService, InstanceLookupService, InstanceLifecycleService, InstanceNaming) is used basically once:

```js
    const instanceNaming = new InstanceNaming();
    this.recipeService = new RecipesService(exec, logger);
    this.instanceLookupService = new InstanceLookupService(exec, instanceNaming, this.deviceRegistry);
    this.instanceLifecycleService = new InstanceLifecycleService(exec, instanceNaming);
    this.deviceQueryHelper = new DeviceQueryHelper(this.recipeService);
    this.allocationHelper = new AllocationHelper(this.instanceLookupService, this.instanceLifecycleService);
```

What is the value of the composition here instead of a more straightforward aggregation? 🙄

Reply:
I disagree. Services can quickly become legacy monoliths if you are not careful. Small classes with concise concerns is the better approach.